### PR TITLE
[WIP] Feature/bind js

### PR DESCRIPTION
**Pull request by GordonPmnt**
This is the very very beginning of JS bind.
Goal is more to show the approach and open PR to contributions.

### CODE REVIEW

File: nodejs_package/brainflow/board-shim.ts [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+    const arch = os.arch();
+    switch (platform) {
+      case 'darwin':
+        return `${this.libPath}/libBoardController.dylib`;
+      case 'win32':
+        return arch === 'x64' ? `${this.libPath}/BoardController.dll` : `${this.libPath}/BoardController32.dll`;
+      case 'linux':
+        return `${this.libPath}/libBoardController.so`;
+      default:
+        throw new Error(`OS ${platform} is not correctly handed by Cortex Desktop App, please contact support.`);
```
Comment (GordonPmnt) [inline nodejs_package/brainflow/board-shim.ts:86]: Oops. Must change this err message here

File: nodejs_package/brainflow/brainflow.types.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+  CLASSIFIER_IS_NOT_PREPARED_ERROR = 21,
+  ANOTHER_CLASSIFIER_IS_PREPARED_ERROR = 22,
+  UNSUPPORTED_CLASSIFIER_AND_METRIC_COMBINATION_ERROR = 23,
+}
+
+// --------------------------- //
+//  Interfaces
+// --------------------------- //
+
+export interface IInputParams {
```
Comment (Andrey1994) [inline nodejs_package/brainflow/brainflow.types.ts:214]: Could you explain a little bit why it should be an interface?

**Review by Andrey1994 — COMMENTED** [review]
looks great so far!
I think it makes sense to complete all methods in BoardShim class and move to DataFilter and MLModel after that.

I can help with docs, examples, CI and cmake scripts to copy libs if you need

**Comment by GordonPmnt** [comment]
> looks great so far! I think it makes sense to complete all methods in BoardShim class and move to DataFilter and MLModel after that.
> 
> I can help with docs, examples, CI and cmake scripts to copy libs if you need

Makes sense indeed. Thanks Andrey.
Your help is more than welcome for scipts, CI, cmake... you have way more experience than me on that. I'll focus on JS transcription.

I also might need some examples / hints about the unit testing approach. :)

File: nodejs_package/brainflow/brainflow.types.ts (hunk 2 above)
Comment (GordonPmnt) [reply nodejs_package/brainflow/brainflow.types.ts:214]: Yes sure. 

In typescript, it is common practice to use `interface` to define the structure of an object, not only for classes.

Some people prefer using `type` instead. I don't have a strong opinion about that to be honest.

File: nodejs_package/brainflow/board-shim.ts [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    const prepared = [0];
+    const res = this.boardController.isPrepared(prepared, this.boardId, this.inputJson);
+    if (res !== BrainFlowExitCodes.STATUS_OK) {
+      throw new Error('Could not check prepared state');
+    }
+    return !!prepared[0];
+  }
+
+  public addStreamer(streamerParams: string, preset = BrainFlowPresets.DEFAULT_PRESET): void {
+    const res = this.boardController.addStreamer(streamerParams, preset, this.boardId, this.inputJson);
```
Comment (GordonPmnt) [inline nodejs_package/brainflow/board-shim.ts:119]: hey @Andrey1994 
In python I see we're doing this:

```py
if streamer_params is None:
            streamer = None
        else:
            try:
                streamer = streamer_params.encode()
            except BaseException:
                streamer = streamer_params
```

I don't see anything similar in Java. Not sure to understand why we're doing this.

**Comment by Andrey1994** [comment]
moved to another PR since here we have some permissions issue

**Pull request closed without merging**